app/src/main/java/com/wednesday/bankingdetails/activity/MainActivity.kt

```diff
@@ -30,8 +44,43 @@ class MainActivity : AppCompatActivity() {
             )
         )
 
-        bankListAdapter = BankListAdapter(bankList)
-        recyclerView.adapter = bankListAdapter
+        getFavouriteBanks()
+    }
+
+    private fun initAdapter(): BankListAdapter {
+        return BankListAdapter(bankList)
+    }
+
+    private fun getFavouriteBanks() {
+        GlobalScope.launch {
+            bankList.clear()
+            bankList.addAll(BankFactory.getMockBanks())
+
+            val data = bankDatabase.favouriteBankDao().getAll()
+
+            data.forEach {
+                bankList.find { bank -> bank.ifsc == it.ifsc }!!.isFavourite = true
```

Comment:
In almost all cases you should never have to force unwrap an optional. This beats the entire purpose of using optional.